docs/Using-Pantheon/Events-and-Logs.md

```diff
@@ -0,0 +1,194 @@
+  	     uint256 public valueA;
+         uint256 public valueB;
+  
+  	     event Event1(uint256 indexed valueA);
+  	     event Event2(uint256 indexed valueB);
+  
+  	     function setValue(uint256 _valueA) public {
+      	    valueA = _valueA;
+      	    emit Event1(_valueA); 
+  	     }
+  	
+  	     function setValueAgain(uint256 _valueB) public {
+      	    valueB = _valueB;
+      	    emit Event2(_valueB); 
+  	     }
+       } 
+    ```
+
+The event signature hash for event 1 is `keccak('Event1(uint256))` and the event signature hash for event 
+2 is `keccak('Event2(uint256))`. The hashes are: 
+
+* `04474795f5b996ff80cb47c148d4c5ccdbe09ef27551820caa9c2f8ed149cce3` for event 1  
+* `06df6fb2d6d0b17a870decb858cc46bf7b69142ab7b9318f7603ed3fd4ad240e` for event 2
+
+!!! tip
+    You can use an online tool such as https://emn178.github.io/online-tools/keccak_256.html to generate 
+    event signature hashes. 
+
+!!! example
+    Log entries from invoking the Solidity contract above:  
+
+    ```json
+    [
+        {
+            "logIndex": "0x0",
+            "removed": false,
+            "blockNumber": "0x84",
+            "blockHash": "0x5fc573d76ec48ec80cbc43f299ebc306a8168112e3a4485c23e84e9a40f5d336",
+            "transactionHash": "0xcb52f02342c2498df82c49ac26b2e91e182155c8b2a2add5b6dc4c249511f85a",
+            "transactionIndex": "0x0",
+            "address": "0x42699a7612a82f1d9c36148af9c77354759b210b",
+            "data": "0x",
+            "topics": [
+                "0x04474795f5b996ff80cb47c148d4c5ccdbe09ef27551820caa9c2f8ed149cce3",
+                "0x0000000000000000000000000000000000000000000000000000000000000001"
+            ]
+        },
+        {
+            "logIndex": "0x0",
+            "removed": false,
+            "blockNumber": "0x87",
+            "blockHash": "0x6643a1e58ad857f727552e4572b837a85b3ca64c4799d085170c707e4dad5255",
+            "transactionHash": "0xa95295fcea7df3b9e47ab95d2dadeb868145719ed9cc0e6c757c8a174e1fcb11",
+            "transactionIndex": "0x0",
+            "address": "0x42699a7612a82f1d9c36148af9c77354759b210b",
+            "data": "0x",
+            "topics": [
+                "0x06df6fb2d6d0b17a870decb858cc46bf7b69142ab7b9318f7603ed3fd4ad240e",
+                "0x0000000000000000000000000000000000000000000000000000000000000002"
+            ]
+        }
+    ]
+    ```
+
+## Topic Filters
+
+[Filter options objects](../Reference/JSON-RPC-API-Objects.md#filter-options-object) have a `topics` key to filter logs by topics. 
+
+Topics are order-dependent. A transaction with a log containing topics `[A, B]` is matched with the following topic filters:
+
+* `[]` - Match any topic
+* `[A]` - Match A in first position 
+* `[[null], [B]]` - Match any topic in first position AND B in second position
+* `[[A],[B]]` - Match A in first position AND B in second position
+* `[[A, B], [C, D]]` - Match (A OR B) in first position AND (C OR D) in second position 
```

Comment:
[A, B] doesn't have C or D in second position...
should this be
[[A, C], [B, D]]
?

Reply:
I double checked this and it's correct as written.  The following topic filter: 
"topics":[
["0xd3610b1c54575b7f4f0dc03d210b8ac55624ae007679b7a928a4f25a709331a8", "0xd3610b1c54575b7f4f0dc03d210b8ac55624ae007679b7a928a4f25a709331a9"], 
["0x0000000000000000000000000000000000000000000000000000000000000001", "0x0000000000000000000000000000000000000000000000000000000000000009"]]

Matches the following topics in a log: 
 "topics": [
   "0xd3610b1c54575b7f4f0dc03d210b8ac55624ae007679b7a928a4f25a709331a8",
   "0x0000000000000000000000000000000000000000000000000000000000000001"]